Sage/carbon
# chore(styled-system-props): add object to prop type for of prop

### Proposed behaviour

Extends the types accepted by the `of` prop in the `StyledSystemProps` component to include `PropTyped.object` as well as there is a warning in the console when viewing http://localhost:9001/?path=/docs/design-system-select--basic

### Current behaviour

The `of` prop accepts types of `PropTypes.node` or `PropTypes.func`

### Checklist



### Additional context


### Testing instructions

Check http://localhost:9001/?path=/docs/design-system-select--basic and confirm the warning has gone
